plugins/modules/cargo.py

```diff
@@ -187,43 +576,92 @@ def main():
         state=dict(default="present", choices=["present", "absent", "latest"]),
         version=dict(default=None, type="str"),
         locked=dict(default=False, type="bool"),
+        features=dict(default=[], type="list", elements="str"),
+        all_features=dict(default=False, type="bool"),
+        force=dict(default=False, type="bool"),
+        directory=dict(default=None, type="path"),
+        git=dict(default=None, type="str"),
+        branch=dict(default=None, type="str"),
+        tag=dict(default=None, type="str"),
+        rev=dict(default=None, type="str"),
+        registry=dict(default=None, type="str"),
+        debug=dict(default=False, type="bool"),
     )
     module = AnsibleModule(argument_spec=arg_spec, supports_check_mode=True)
 
     name = module.params["name"]
     state = module.params["state"]
     version = module.params["version"]
+    force = module.params["force"]
+    features = module.params["features"]
+    all_features = module.params["all_features"]
+    directory = module.params["directory"]
+    registry = module.params["registry"]
+    git = module.params["git"]
+    branch = module.params["branch"]
+    tag = module.params["tag"]
+    rev = module.params["rev"]
 
     if not name:
         module.fail_json(msg="Package name must be specified")
 
+    if len(name) > 1 and features:
+        module.fail_json(
+            msg="Cannot determine which features are for what package for multiple packages"
+        )
+
+    if len(name) > 1 and version:
+        module.fail_json(msg='Instead of version, use "crate@version"...')
+
+    if version and (directory or git or branch or tag or rev):
+        module.fail_json(msg="version option only works with a registry")
+
+    if features and all_features:
+        module.fail_json(msg="features list is a redundant with all_features")
+
+    if version is not None and (git is not None or directory is not None):
+        module.fail_json(msg="version option can only be used with a cargo registry")
+
+    if sum(1 for x in [git, directory, registry] if x is not None) > 1:
+        module.fail_json(msg="Only one of git, registry or directory is allowed")
```

Comment:
Such checks should be done with `mutually_exclusive` in the `AnsibleModule` call.

Reply:
I saw `mutually_exclusive` raises a `TypeError` if not satisfied. So do I have to use a `try-except` block instead of this `if` statement?

Reply:
`mutually_exclusive` is handled by AnsibleModule. You do not have to call it yourself, or handle any exception. AnsibleModule does that. Simply pass `mutually_exclusive=[...]` to the `AnsibleModule()` constructor call.